src/Test/Perf/infra/monitor_and_start_test.csx

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+Task.Run(() => StartWatching());
+Console.WriteLine("Press any key to continue");
+Console.Read();
+
+void StartWatching()
+{
+    FileSystemWatcher watcher = new FileSystemWatcher();
+    watcher.Path = Args[0];
+    watcher.Created += new FileSystemEventHandler(OnChanged);
+    watcher.IncludeSubdirectories = false;
+    watcher.EnableRaisingEvents = true;
+    while (true)
+    {
+        // Stay in loop
+    }
+}
+
+void OnChanged(object sender, FileSystemEventArgs e)
+{
+    // Wait for 120 seconds to make sure all the contents are copied over to the share
+    // If there is a better estimate then we can modify the wait time to be a little over the estimate.
+    Thread.Sleep(120000);
```

Comment:
I would assume that the build produces the binaries and keeps it in the build machine until all of the build process is done. Once the build process is done, we copy everything over to the share. So above approach wont work